modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/helper/DefinitionParserHelper.java

```diff
@@ -166,37 +166,47 @@ public static void addTable(TableDefinition tableDefinition, ConcurrentMap<Strin
                 if (tableRef != null) {
                     Map<String, String> storeConfigs = siddhiAppContext.getSiddhiContext().getConfigManager()
                             .extractStoreConfigs(tableRef);
-                    final String tableTypeFromRef = storeConfigs.get(SiddhiConstants.ANNOTATION_ELEMENT_TYPE);
-
-                    if (tableTypeFromRef == null) {
-                        throw new SiddhiAppCreationException("Table type must be defined in the store element of " +
-                                "name '" + tableRef + "' in the deployment.yaml file.");
+                    if (storeConfigs.size() == 0) {
```

Comment:
Is there be a possibility to storeConfigs getting 'null' if you extractStoreConfigs ?